lib/admin_plugins/cleanstatusdb.js

```diff
@@ -67,3 +75,58 @@ cleanstatusdb.actions[0].code =  function deleteRecords(client, callback) {
     }
   });
 };
+
+cleanstatusdb.actions[1].init = function init(client, callback) {
+  var translate = client.translate;
+  var $status = $('#admin_' + cleanstatusdb.name + '_1_status');
+
+  $status.hide();
+
+  var numDays =  '<br/>'
+      + '<label for="admin_devicestatus_days">'
+      + translate('Number of Days to Keep:')
+      + '  <input id="admin_devicestatus_days" value="30" size="3" min="1" max="365"/>'
+      + '</label>';
+
+  $('#admin_' + cleanstatusdb.name + '_1_html').html(numDays);
+
+  if (callback) { callback(); }
+};
+
+cleanstatusdb.actions[1].code =  function deleteOldRecords(client, callback) {
+  var translate = client.translate;
+  var $status = $('#admin_' + cleanstatusdb.name + '_1_status');
+  var numDays = Number($('#admin_devicestatus_days').val());
+
+  if (isNaN(numDays) || (numDays <= 0)) {
```

Comment:
The numDays check doesn't match 1..365 in the range in the HTML input tag (zero is allowed and 366 also). 
I think a range of 1 to 365 should be checked in the the input, in the javacript and also on the API-side

Reply:
@PieterGit, the API side is based on the generic find capabilities in the mongodb api.  After looking at this again, I thought it best not to artificially limit the maximum number of days the user could choose to keep.

If the user has a paid version of mLab or some other hosting solution that enables to to maintain a much larger database, they may want to keep 2 years, 3 years, etc.